src/o365/spo/commands/feature/feature-list.ts

```diff
@@ -0,0 +1,175 @@
+    return telemetryProps;
+  }
+
+  public commandAction(cmd: CommandInstance, args: CommandArgs, cb: () => void): void {
+    const scope: string = (args.options.scope) ? args.options.scope : 'Web';
+    const resource: string = Auth.getResourceFromUrl(args.options.url);
+    let siteAccessToken: string = '';
+
+    auth.getAccessToken(resource, auth.service.refreshToken as string, cmd, this.debug)
+      .then((accessToken: string): request.RequestPromise => {
+        siteAccessToken = accessToken;
+
+        if (this.debug) {
+          cmd.log(`Retrieved access token ${accessToken}. Retrieving request digest...`);
+        }
+
+        return this.getRequestDigestForSite(args.options.url, accessToken, cmd, this.debug);
+      })
+      .then((contextResponse: ContextInfo): request.RequestPromise => {
+        if (this.debug) {
+          cmd.log('Response:');
+          cmd.log(JSON.stringify(contextResponse));
+          cmd.log('');
+
+          cmd.log(`Attempt to get feature list with scope: ${scope}`);
+          cmd.log('');
+        }
+
+        const requestOptions: any = {
+          url: `${args.options.url}/_api/${scope}/Features?$select=DisplayName,DefinitionId`,
+          headers: Utils.getRequestHeaders({
+            authorization: `Bearer ${siteAccessToken}`,
+            accept: 'application/json;odata=nometadata'
+          }),
+          json: true
+        };
+
+        return request.get(requestOptions);
+      })
+      .then((features: { value: Feature[] }): void => {
+        if (this.debug) {
+          cmd.log('Response:');
+          cmd.log(features);
+          cmd.log('');
+        }
+
+        if (features.value && features.value.length > 0) {
+          if (args.options.output === 'json') {
+            cmd.log(features.value);
+          }
+          else {
+            cmd.log(features.value.map(f => {
+              return {
+                DefinitionId: f.DefinitionId,
+                DisplayName: f.DisplayName
+              };
+            }));
+          }
+        }
+        else {
+          if (this.verbose) {
+            cmd.log('No features found');
+          }
+        }
+        cb();
+      }, (err: any): void => this.handleRejectedPromise(err, cmd, cb));
+  }
+
+  public options(): CommandOption[] {
+    const options: CommandOption[] = [
+      {
+        option: '-u, --url <url>',
+        description: 'Url of the site (collection) to retrieve the feature from'
+      },
+      {
+        option: '-s, --scope [scope]',
+        description: 'Scope of the feature. Allowed values Site|Web. Default Web',
+        autocomplete: ['Site', 'Web']
+      }
+    ];
+
+    const parentOptions: CommandOption[] = super.options();
+    return options.concat(parentOptions);
+  }
+
+  public validate(): CommandValidate {
+    return (args: CommandArgs): boolean | string => {
+
+      if (SpoCommand.isValidSharePointUrl(args.options.url) !== true) {
+        return 'Missing required option url';
```

Comment:
This error message would be confusing when an invalid URL would be specified. The `isValidSharePointUrl` method already returns an error message, so we should return it directly, rather than defining a new message.